server/src/main/resources/rules/rules.js

```diff
@@ -308,7 +308,7 @@ function createPool(pool, consumer) {
         }
         else {
             if (attribute === 'addons' || attribute === 'roles') {
-                poolSet = this.getProductAttribute(attribute).split(',');
+                poolSet = this.getProductAttribute(attribute).trim().split(/\s*,\s*/);
```

Comment:
We should probably use the same regex here that also parses out empty/whitespaced comma pairs which we're using above.

Reply:
Done. Updated to handle whitespaced commas.